volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -36,6 +35,30 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._mod_mem_type = None  # Initialize _mod_mem_type to None for memoization
 
+    def is_valid(self):
+        layer = self._context.layers[self.vol.layer_name]
+        # Make sure the entire module content is readable
+        if not layer.is_valid(self.vol.offset, self.vol.size):
+            return False
+
+        core_size = self.get_core_size()
+        if not (
+            1 <= core_size <= 20000000
```

Comment:
These hardcoded limits come about from...?  For arbitrary values like 20000000, I'd prefer that they were stashed in `constants.linux` with a brief comment to describe them, so that they're consistent and can be changed all at once if needed.

Reply:
yeah, see vol2

Reply:
done